### DISCUSSION TITLE
dnscrypt-proxy vs DoH Profile

### DISCUSSION THREAD

**Discussion by bart3nder** [DNSCrypt/dnscrypt-proxy#2476] [Q&A] [answered]
Hi! Thanks for the great project.
A little question, What is the difference between using dnscrypt-proxy and set Mac DNS to `127.0.0.1:53`, and Using DoH profiles (Like from https://github.com/paulmillr/encrypted-dns)? In terms of performance, stability or security.
Thank you!

**Comment by lifenjoiner** [comment, marked as answer]
The essential difference is using a local proxy Vs using remote DoH directly. Resulting:
1. With a proxy, you can do some control, including ad blocking, of course costs some time, while in direct, all depends on the remote server.
2. `dnscrypt-proxy` has dozens of upstream servers, does quality ranking, and adopts one dynamically, while specific DoH is solid.
3. Performance, stability and security are all complicated things, and may be different server to server, time to time, and ISP to ISP. Even, if a server claims it is anonymous, no log, no block etc., will you trust it 100%? Anyway, `dnscrypt-proxy` has one more feature: Anonymized DNS with relays.

**Reply by bart3nder** [reply]
Thanks for the detailed answer, Appreciate it.
